src/carousel.scss

```diff
@@ -0,0 +1,155 @@
+@import "./new-settings";
+@import "./mixins";
+
+$block: #{$fd-namespace}-carousel;
+$button: #{$fd-namespace}-button;
+
+.#{$block} {
+  $fd-carousel-page-indicator-border: 0.0625rem solid var(--sapPageFooter_BorderColor) !default;
+  $fd-carousel-button-size: 2.125rem !default;
+  $fd-carousel-button-content-offset: 0.5rem !default;
+  $fd-carousel-dot-size: 0.25rem !default;
+  $fd-carousel-dot-active-size: 0.5rem !default;
+
+  @include fd-reset();
+  @include fd-flex(column);
+
+  width: 100%;
+  min-width: 15.5rem;
+  max-width: 100%;
+  height: 100%;
+
+  &__content {
+    @include fd-reset();
+
+    flex: 1 1 auto;
+    width: 100%;
+    background: var(--sapBackgroundColor);
+    position: relative;
+
+    .#{$block}__button.#{$button} {
+      box-shadow: var(--sapContent_Shadow1);
+      position: absolute;
+      top: calc(50% - 1.0625rem);
+      margin: 0;
+
+      &.sap-icon--slim-arrow-left {
+        left: $fd-carousel-button-content-offset;
+
+        @include fd-rtl() {
+          left: initial;
+          right: $fd-carousel-button-content-offset;
+        }
+      }
+
+      &.sap-icon--slim-arrow-right {
+        right: $fd-carousel-button-content-offset;
+
+        @include fd-rtl() {
+          right: initial;
+          left: $fd-carousel-button-content-offset;
+        }
+      }
+    }
+  }
+
+  &__item {
+    @include fd-reset();
+    @include fd-flex-center();
+
+    width: 100%;
+    height: 100%;
+    display: none;
+
+    &--active {
+      display: block;
+    }
+  }
+
+  &__page-indicator-container {
+    @include fd-reset();
+    @include fd-flex-center();
+
+    width: 100%;
+    height: 100%;
+    min-height: 2.75rem;
+    max-height: 3.5rem;
+    background: var(--sapPageFooter_Background);
+    padding: 0.375rem 0.5rem;
+    border-top: $fd-carousel-page-indicator-border;
+
+    &:first-child {
+      border-top: none;
+      border-bottom: $fd-carousel-page-indicator-border;
+    }
+  }
+
+  &__page-indicators {
+    @include fd-reset();
+
+    @include fd-flex-center() {
+      flex-wrap: wrap;
+    }
+
+    width: 9rem;
+    padding: 0.5rem;
+  }
+
+  &__page-indicator {
+    @include fd-reset();
+
+    list-style-type: none;
+    margin: 0 0.375rem;
+    width: $fd-carousel-dot-size;
+    height: $fd-carousel-dot-size;
+    border-radius: 50%;
+    background-color: var(--sapContent_NonInteractiveIconColor);
+
+    &--active {
+      margin: 0 0.25rem;
+      width: $fd-carousel-dot-active-size;
+      height: $fd-carousel-dot-active-size;
+      background-color: var(--sapSelectedColor);
+    }
+  }
+
+  &__text {
+    @include fd-reset();
+    @include fd-ellipsis();
+
+    color: var(--sapPageFooter_TextColor);
+    text-align: center;
+  }
+
+  &__button.#{$button} {
+    @include fd-flex-center();
+    @include set-width($fd-carousel-button-size);
+    @include set-height($fd-carousel-button-size);
+
+    border-radius: 50%;
+    padding: 0;
+    box-shadow: none;
+    margin: 0.25rem;
```

Comment:
Maybe let's add additional clickable area as additional spacing for buttons. It's done in #1305  button refactor as pseudo element.

Reply:
Yes, that's why I added the margin now so the spacing is correct. Once we have the refactored button the margin will be removed, the spacing will be correct and we will have the clickable area.